src/app.js

```diff
@@ -26,52 +26,55 @@ function main(sources) {
     time: new Date()
   }]);
 
+  const intent = DOM => {
+    const changeMessage$ = sources.DOM
+      .select('.input-message')
+      .events('input')
+      .map(e => e.target.value)
+      .share();
 
-
-  const username$ = sources.DOM
-    .select('.swish-input')
-    .events('change')
-    .startWith({ target: { value: '' } })
-    .map(e => e.target.value);
-
-  const usernameSubmit$ = sources.DOM.select('.username-form').events('submit');
-
-  const usernameChanges$ = Observable.combineLatest(
-    username$,
-    usernameSubmit$,
-    (username, submitEvent) => username
-  ).startWith('');
-
-
-  const state$ = Observable.combineLatest(
-    allPusherMessages$,
-    usernameChanges$,
-    (pusherMessages, username) => ({ pusherMessages, username })
-  );
-
-  const inputValue$ = sources.DOM
-    .select('.input-message')
-    .events('change')
-    .startWith({ target: { value: '' } })
-    .map(e => e.target.value)
-
-  const messageSubmits$ = sources.DOM.select('.messages-form').events('submit');
-
-  const sendClicks$ = sources.DOM.select('.send-message').events('click');
-
-  const clickOrSubmit$ = Observable.merge(messageSubmits$, sendClicks$)
-
-  const request$ = Observable.combineLatest(
-    clickOrSubmit$,
-    inputValue$,
-    usernameChanges$,
-    (submit, inputVal, username) => ({ inputVal, username })
-    // TODO: this debounce stops duplicate messages - which I need to figure out why they happen
-    // I think that listening to clicks & input value changes means we get an event when either one of those things happen
-    // maybe?
-  ).debounce(10).filter(
-    ({ inputVal }) => inputVal !== ''
-  ).map(({ inputVal, username }) => {
+    return {
+      submitUsername$: sources.DOM
+        .select('.swish-input')
+        .events('input')
+        .sample(sources.DOM.select('.username-form').events('submit'))
```

Comment:
is it `sample` here that solves the dup request? If I'm reading this right it means we take the value from the `swish-input`  but only when we get a submit event?